core/src/main/java/org/mule/runtime/core/internal/connection/PoolingConnectionManagementStrategy.java

```diff
@@ -79,6 +83,8 @@
 
   private C borrowConnection() throws Exception {
     C connection = pool.borrowObject();
+    LOGGER.trace("Borrowed connection {} from the pool {}", connection.toString(), poolId);
```

Comment:
Same comment regarding log level, I think debug is the appropriate for all the pool logs.